sdk/communication/communication-chat/test/public/utils/recordedClient.ts

```diff
@@ -15,55 +18,63 @@ import {
 import { CommunicationIdentityClient, CommunicationUserToken } from "@azure/communication-identity";
 import { generateToken } from "./connectionUtils";
 
-if (isNode) {
-  dotenv.config();
-}
-
 export interface RecordedClient {
   chatClient: ChatClient;
   recorder: Recorder;
 }
 
-const replaceableVariables: { [k: string]: string } = {
+const envSetupForPlayback: { [k: string]: string } = {
   COMMUNICATION_LIVETEST_DYNAMIC_CONNECTION_STRING: "endpoint=https://endpoint/;accesskey=banana",
 };
 
-export const environmentSetup: RecorderEnvironmentSetup = {
-  replaceableVariables,
-  customizationsOnRecordings: [
-    (recording: string): string => recording.replace(/"token"\s?:\s?"[^"]*"/g, `"token":"token"`),
-    (recording: string): string => recording.replace(/(https:\/\/)([^/',]*)/, "$1endpoint"),
-    (recording: string): string => recording.replace("endpoint:443", "endpoint"),
-  ],
-  queryParametersToSkip: [],
+const fakeToken = generateToken();
+export const recorderOptions: RecorderStartOptions = {
+  envSetupForPlayback,
+  sanitizerOptions: {
+    connectionStringSanitizers: [
+      {
+        fakeConnString: envSetupForPlayback["COMMUNICATION_LIVETEST_DYNAMIC_CONNECTION_STRING"],
+        actualConnString: env["COMMUNICATION_LIVETEST_DYNAMIC_CONNECTION_STRING"] || undefined,
+      },
+    ],
+    bodyKeySanitizers: [{ jsonPath: "$.accessToken.token", value: fakeToken }],
+  },
 };
 
-export async function createTestUser(): Promise<CommunicationUserToken> {
+export async function createTestUser(recorder: Recorder): Promise<CommunicationUserToken> {
   const identityClient = new CommunicationIdentityClient(
-    env.COMMUNICATION_LIVETEST_DYNAMIC_CONNECTION_STRING
+    assertEnvironmentVariable("COMMUNICATION_LIVETEST_DYNAMIC_CONNECTION_STRING"),
+    recorder.configureClientOptionsCoreV1({})
```

Comment:
Isn't Chat CoreV2? It's using `core-rest-pipeline` at least.

Reply:
`communication-identity` is core-v1 looks like.